evm_arithmetization/src/verifier.rs

```diff
@@ -116,13 +116,32 @@ fn verify_initial_memory<
     Ok(())
 }
 
-pub fn verify_proof<F: RichField + Extendable<D>, C: GenericConfig<D, F = F>, const D: usize>(
+pub fn verify_all_proofs<
```

Comment:
I'd rather put this in a `testing` module, similarly to `prove_all_segments` as it isn't aimed to be used outside of testing.